samcli/cli/main.py

```diff
@@ -58,6 +59,18 @@ def print_info(ctx, param, value):
     ctx.exit()
 
 
+def print_version_option(ctx, param, value):
+    """
+    Since default implementation for click.version_option doesn't support callback's we need to create a simple version
+    of it, which will print version information, and check for new version and exit the cli
+    """
+    if not value or ctx.resilient_parsing:
+        return
+    click.echo(f"SAM CLI, version {__version__}", color=ctx.color)
+    inform_newer_version(force_check=True)
```

Comment:
One thing to keep in mind is that `--version` (for pretty much every command) is often used as a dead-simple, immediate sanity check (e.g. in CI/CD installing SAM CLI and running `sam --version` to make sure it doesn't return a zero exit code). I don't think that forcing network operations at every call is sensible, as people will often except immediate feedback, and the implicit assumption seems to be that `--version` barely prints a constant and exits.

Reply:
This was coming from UX team as a suggestion, if we know that is been used in this way, I agree to remove this force check for `--version`.

Reply:
Agreed that it will create too much burden for a command which is usually used as sanity check.